src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const array = [];
+  const stateClone = { ...state };
+
+  for (const key of actions) {
```

Comment:
[CODE STYLE]: The variable 'key' in the for...of loop is misleading since it represents an action, not a key. Consider renaming it to 'action' for clarity.